tests/uts/test_agent.py

```diff
@@ -0,0 +1,64 @@
+import json
+import logging
+import os
+import unittest
+import uuid
+from threading import Thread
+
+from fakeredis import FakeStrictRedis
+from mock import patch
+
+from agent.talker import TalkerAgent
+
+
+def get_uuid():
+    return uuid.uuid4().__str__()
+
+
+def kill_process_before_signaling(self, job_id, signal):
+    job = self.current_processes.get(job_id)
+    os.killpg(os.getpgid(job.popen.pid), signal)
+    while job.popen.poll() is None:
+        pass
+    job.send_signal(signal)
+
+
+JOBS_DIR = '/tmp/talker/jobs'
+JOBS_SEEN = '/tmp/talker/jobs/eos.json'
```

Comment:
I would expect that only patching `JOBS_DIR` will be sufficient. let's fix that in the code